lib/active_set.rb

```diff
@@ -41,27 +44,26 @@ def method_missing(method_name, *args, &block)
   def respond_to_missing?(method_name, include_private = false)
     @view.respond_to?(method_name) || super
   end
+  # :nocov:
 
-  def filter(instructions)
-    filterer = Processor::Filter.new(@view, instructions)
-    reinitialize(filterer.process, :filter, instructions)
+  def filter(instructions_hash)
+    filterer = Filtering::Operation.new(@view, instructions_hash)
+    reinitialize(filterer.execute, :filter, filterer.operation_instructions)
```

Comment:
The idea with the `Operation` classes exposing an `operation_instructions` method is to allow them to set defaults or munge the incoming instructions in any way, so that the `@instructions` hash attached to the ActiveSet instance has a meaningful representation of the operation instructions that have been applied. This is particularly needed for pagination, where the operation has defaults (for `size` and `page`)